src/SpeedCalc.cc

```diff
@@ -84,6 +84,29 @@ int SpeedCalc::calculateSpeed()
   return speed;
 }
 
+int SpeedCalc::calculateNewestSpeed(int seconds)
+{
+  const auto& now = global::wallclock();
+  removeStaleTimeSlot(now);
+
+  int64_t bytesCount(0);
+  auto it=timeSlots_.rbegin();
+  while (it != timeSlots_.rend()) {
+    if (it->first.difference(now) > seconds*1_s) {
+      break;
+    }
+    bytesCount += (*it++).second;
+  }
+  if (it == timeSlots_.rbegin()) return 0;
```

Comment:
i'm not sure to understand. 
all slots that have a timetag in the range are considered.
the 'if' on rbegin is there in case the timeSlots_ list is empty or if the while loop exits immediately (it is actually unlikely but one never knows). Note that the while loop exits normally with one position ahead on the iterator, hence if the iterator has not moved from its first rbegin position, then we are in this case and we have no slot to compute the speed upon.
please comment back if i missed something, thanks